packages/docusaurus-theme-classic/src/theme/BackToTopButton/index.tsx

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useState} from 'react';
+import clsx from 'clsx';
+import useScrollPosition from '@theme/hooks/useScrollPosition';
+
+import styles from './styles.module.css';
+
+const threshold = 300;
+
+function smoothScrollToTop() {
+  const currentScroll = document.documentElement.scrollTop;
+
+  if (currentScroll > 0) {
+    requestAnimationFrame(smoothScrollToTop);
```

Comment:
Is this recursion solution for Safari? 
Where does it come from?

What about using the native `scrollTo(0,{behavior: "smooth"})` for other browsers?

Apparently we can test for support with `'scrollBehavior' in document.documentElement.style`


Also the recursion should stop if user attempts to scroll down again, otherwise the user scroll and the current animation conflict 🤪

Reply:
No, this is such cross-browser generic way to make smooth scrolling, since the call `scrollTo(0,{behavior: "smooth"}` with `scroll-behavior` does not guarantee it.